exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/QueryResultHandler.java

```diff
@@ -278,49 +297,31 @@ public void queryIdArrived(QueryId queryId) {
 
   private class SubmissionListener extends BaseRpcOutcomeListener<QueryId> {
     private final UserResultsListener resultsListener;
-    private final RemoteConnection connection;
-    private final ChannelFuture closeFuture;
-    private final ChannelClosedListener closeListener;
     private final AtomicBoolean isTerminal = new AtomicBoolean(false);
 
-    public SubmissionListener(RemoteConnection connection, UserResultsListener resultsListener) {
-      super();
+    public SubmissionListener(UserResultsListener resultsListener) {
       this.resultsListener = resultsListener;
-      this.connection = connection;
-      this.closeFuture = connection.getChannel().closeFuture();
-      this.closeListener = new ChannelClosedListener();
-      closeFuture.addListener(closeListener);
-    }
-
-    private class ChannelClosedListener implements GenericFutureListener<Future<Void>> {
-
-      @Override
-      public void operationComplete(Future<Void> future) throws Exception {
-        resultsListener.submissionFailed(UserException.connectionError()
-            .message("Connection %s closed unexpectedly.", connection.getName())
-            .build(logger));
-      }
-
     }
 
     @Override
     public void failed(RpcException ex) {
       if (!isTerminal.compareAndSet(false, true)) {
+        logger.warn("Received multiple outcomes (success/failure) while submitting query.");
         return;
       }
 
-      closeFuture.removeListener(closeListener);
-      resultsListener.submissionFailed(UserException.systemError(ex).build(logger));
-
+      resultsListener.submissionFailed(UserException.systemError(ex)
+          .message("Unexpected failure while submitting query to a Drillbit.")
```

Comment:
This will override the original exception's message. You should either include it in the error message, or if you think it's not useful to the end user, log it so dev can find what wrong